src/framework/components/camera/component.js

```diff
@@ -104,13 +109,74 @@ class CameraComponent extends Component {
     constructor(system, entity) {
         super(system, entity);
 
-        this._camera = new Camera();
         this._camera.node = entity;
 
         // postprocessing management
         this._postEffects = new PostEffectQueue(system.app, this);
     }
 
+    /**
+     * Sets the name of the shader pass the camera will use when rendering.
+     *
+     * @param {string} name - The name of the shader pass. Defaults to undefined, which is
+     * equivalent to {@link SHADERTYPE_FORWARD}. Can be:
+     *
+     * - {@link SHADERTYPE_FORWARD}
+     * - {@link SHADERTYPE_DEBUG_ALBEDO}
+     * - {@link SHADERTYPE_DEBUG_OPACITY}
+     * - {@link SHADERTYPE_DEBUG_WORLDNORMAL}
+     * - {@link SHADERTYPE_DEBUG_SPECULARITY}
+     * - {@link SHADERTYPE_DEBUG_GLOSS}
+     * - {@link SHADERTYPE_DEBUG_METALNESS}
+     * - {@link SHADERTYPE_DEBUG_AO}
+     * - {@link SHADERTYPE_DEBUG_EMISSION}
+     * - {@link SHADERTYPE_DEBUG_LIGHTING}
+     * - {@link SHADERTYPE_DEBUG_UV0}
+     *
+     * Additionally, a new name can be specified, which creates a new shader pass with the given
+     * name. The name provided can only use alphanumeric characters and underscores. When a shader
+     * is compiled for the new pass, a define is added to the shader. For example, if the name is
+     * 'custom_rendering', the define 'CUSTOM_RENDERING_PASS' is added to the shader, allowing the
+     * shader code to conditionally execute code only when that shader pass is active.
+     *
+     * Another instance where this approach may prove useful is when a camera needs to render a more
+     * cost-effective version of shaders, such as when creating a reflection texture. To accomplish
+     * this, a callback on the material that triggers during shader compilation can be used. This
+     * callback can modify the shader generation options specifically for this shader pass.
+     *
+     * ```javascript
+     * var shaderPassId = camera.setShaderPassName('custom_rendering');
+     *
+     * material.onUpdateShader = function (options) {
+     *    if (options.pass === shaderPassId) {
+     *        options.litOptions.normalMapEnabled = false;
+     *        options.litOptions.useSpecular = false;
+     *    }
+     *    return options;
+     * };
+     * ```
```

Comment:
This should be in an `@example` block.

Reply:
And don't use `var`. 😉

Reply:
When I use example block, the example is moved to the start which I don't like. Mesh class does the same to keep it ordered. Is there some other way?
![Screenshot 2023-04-21 at 15 24 06](https://user-images.githubusercontent.com/59932779/233660999-9c14e36c-f0a4-4a87-80e5-8b9c2be233ce.png)